driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -78,7 +78,8 @@ public final class MongoClientSettings {
                     new BsonCodecProvider(),
                     new EnumCodecProvider(),
                     new ExpressionCodecProvider(),
-                    new Jep395RecordCodecProvider()));
+                    new Jep395RecordCodecProvider(),
+                    new KotlinDataClassCodecProvider()));
```

Comment:
There is an issue here for the migration of kmongo: this is not backward compatible - and it prevents to add easily other kotlin codecs for data class. I had to define my own default settings in kmongo because of an issue with EnumCodec (see https://github.com/Litote/kmongo/blob/master/kmongo-shared/src/main/kotlin/org/litote/kmongo/util/KMongoUtil.kt#L110 )

I would be great to add a way to configure the list of default CodecProviders (ServiceLoader ?) 

By the way, you forget to update the javadoc of #getDefaultCodecRegistry()

Reply:
Extending codec registries is supported, either via `MongoClientSettings` or using `withCodecRegistry`.  As they are registries, they are order dependent so users typically do something like:  `fromRegistries(fromCodecs(myCustomCodec), DefaultCodecRegsitry)`

Will update the javadoc.

Reply:
Ok thank you. The issue with kmongo is that I have a codec to add at the end of the list of default providers - but I can manage that by removing EnumCodecProvider & KotlinDataClassCodecProvider ;)